packages/core/core/src/Asset.js

```diff
@@ -101,38 +93,38 @@ export default class Asset {
     this.cache = options.cache;
     this.map = options.map;
     this.mapKey = options.mapKey;
-    this.dependencies = options.dependencies
-      ? new Map(options.dependencies)
-      : new Map();
-    this.connectedFiles = options.connectedFiles
-      ? new Map(options.connectedFiles)
-      : new Map();
+    this.dependencies = options.dependencies || new Map();
+    this.connectedFiles = options.connectedFiles || new Map();
     this.outputHash = options.outputHash || '';
     this.env = options.env;
     this.meta = options.meta || {};
     this.stats = options.stats;
-    this.symbols = new Map(options.symbols || []);
+    this.symbols = options.symbols || new Map();
     this.sideEffects = options.sideEffects != null ? options.sideEffects : true;
   }
 
-  serialize(): SerializedOptions {
+  static deserialize(opts: AssetOptions) {
```

Comment:
is this still necessary?

Reply:
unfortunately yes if we want to exclude `code`, `map`, and `ast` from the cache